Error when trying out User Managed Access with WSO2 Identity Server flow

**Describe the issue:**
When following the documentation on [User Managed Access with WSO2 Identity Server 
](https://is.docs.wso2.com/en/6.0.0/guides/access-delegation/uma/) and trying to register the resource instead of the expected response get the below error,  

`ERROR {org.wso2.carbon.identity.oauth.uma.resource.endpoint.impl.ResourceRegistrationApiServiceImpl} - Error when registering resource for client: PhQIdnyaKLKhYmLcD0PbDLMfcbga org.wso2.carbon.identity.oauth.uma.common.exception.UMAServerException: Server error occurred while registering resource.
	at org.wso2.carbon.identity.oauth.uma.resource.service.dao.ResourceDAO.registerResource(ResourceDAO.java:96)
	at org.wso2.carbon.identity.oauth.uma.resource.service.impl.ResourceServiceImpl.registerResource(ResourceServiceImpl.java:40)
	at org.wso2.carbon.identity.oauth.uma.resource.endpoint.impl.ResourceRegistrationApiServiceImpl.registerResource(ResourceRegistrationApiServiceImpl.java:94)
	at org.wso2.carbon.identity.oauth.uma.resource.endpoint.ResourceRegistrationApi.registerResource(ResourceRegistrationApi.java:98)
	at jdk.internal.reflect.GeneratedMethodAccessor263.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179)
	at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96)
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201)
	at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104)
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59)
	at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307)
	at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121)
	at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265)
	at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234)
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208)
	at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160)
	at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225)
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304)
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPost(AbstractHTTPServlet.java:217)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681)
	at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:540)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.wso2.carbon.identity.context.rewrite.valve.TenantContextRewriteValve.invoke(TenantContextRewriteValve.java:119)
	at org.wso2.carbon.identity.context.rewrite.valve.OrganizationContextRewriteValve.invoke(OrganizationContextRewriteValve.java:112)
	at org.wso2.carbon.tomcat.ext.valves.SameSiteCookieValve.invoke(SameSiteCookieValve.java:38)
	at org.wso2.carbon.identity.cors.valve.CORSValve.invoke(CORSValve.java:89)
	at org.wso2.carbon.identity.authz.valve.AuthorizationValve.invoke(AuthorizationValve.java:136)
	at org.wso2.carbon.identity.auth.valve.AuthenticationValve.invoke(AuthenticationValve.java:134)
	at org.wso2.carbon.tomcat.ext.valves.CompositeValve.continueInvocation(CompositeValve.java:106)
	at org.wso2.carbon.tomcat.ext.valves.TomcatValveContainer.invokeValves(TomcatValveContainer.java:49)
	at org.wso2.carbon.tomcat.ext.valves.CompositeValve.invoke(CompositeValve.java:67)
	at org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve.invoke(CarbonStuckThreadDetectionValve.java:152)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:687)
	at org.wso2.carbon.tomcat.ext.valves.CarbonContextCreatorValve.invoke(CarbonContextCreatorValve.java:63)
	at org.wso2.carbon.tomcat.ext.valves.RequestEncodingValve.invoke(RequestEncodingValve.java:49)
	at org.wso2.carbon.tomcat.ext.valves.RequestCorrelationIdValve.invoke(RequestCorrelationIdValve.java:137)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:359)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:399)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:889)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1735)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException: Could not commit the transaction.
	at org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate.withTransaction(NamedJdbcTemplate.java:105)
	at org.wso2.carbon.identity.oauth.uma.resource.service.dao.ResourceDAO.registerResource(ResourceDAO.java:70)
	... 58 more
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: '%s.' INSERT INTO IDN_UMA_RESOURCE (RESOURCE_ID, RESOURCE_NAME, TIME_CREATED, RESOURCE_OWNER_NAME, CLIENT_ID, TENANT_ID, USER_DOMAIN) VALUES (:resource_id;,:resource_name;,:time_created;,:resource_owner_name;,:client_id;,:tenant_id;,:user_domain;)
	at org.wso2.carbon.database.utils.jdbc.NamedTemplate.executeInsert(NamedTemplate.java:80)
	at org.wso2.carbon.identity.oauth.uma.resource.service.dao.ResourceDAO.lambda$registerResource$1(ResourceDAO.java:72)
	at org.wso2.carbon.database.utils.jdbc.NamedJdbcTemplate.withTransaction(NamedJdbcTemplate.java:81)
	... 59 more
Caused by: java.sql.SQLException: Creating the record failed with Auto-Generated ID, no ID obtained.
	at org.wso2.carbon.database.utils.jdbc.NamedTemplate.executeInsert(NamedTemplate.java:66)
	... 61 more
`

**How to reproduce:**
Follow the flow in [documentation](https://is.docs.wso2.com/en/6.0.0/guides/access-delegation/uma/) and register a resource.

**Expected behavior:**
Expected response of containing an id should be received as mention in the documentation.

**Environment information** (_Please complete the following information; remove  any unnecessary fields_) **:**
 - Product Version: IS 6.0.0-alpha3
 - OS: MacOS 12.4
 - JDK: Oracle JDK 11.0.15

--- a/components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/NamedTemplate.java
+++ b/components/org.wso2.carbon.database.utils/src/main/java/org/wso2/carbon/database/utils/jdbc/NamedTemplate.java
@@ -39,6 +39,8 @@ public class NamedTemplate<T> {
 
     private static final Logger log = LoggerFactory.getLogger(NamedTemplate.class);
 
+    private static final String ID = "ID";
+
     /**
      * @param query            The SQL for insert/update.
      * @param namedQueryFilter Query filter to named prepared statement parameter binding.
@@ -53,7 +55,8 @@ public class NamedTemplate<T> {
         int resultId;
         try {
             if (fetchInsertedId) {
-                try (NamedPreparedStatement namedPreparedStatement = new NamedPreparedStatement(connection, query)) {
+                try (NamedPreparedStatement namedPreparedStatement = new NamedPreparedStatement(connection, query,
+                        ID)) {
                     doInternalUpdate(namedQueryFilter, namedPreparedStatement);
                     logDebug("Mapping generated key (Auto Increment ID) to the object");
                     try (ResultSet generatedKeys = namedPreparedStatement.getGeneratedKeys()) {

